packages/core/src/application_ref.ts

```diff
@@ -523,8 +535,11 @@ export class PlatformRef {
     this._modules.slice().forEach(module => module.destroy());
     this._destroyListeners.forEach(listener => listener());
 
-    const destroyListener = this._injector.get(PLATFORM_ON_DESTROY, null);
-    destroyListener?.();
+    const destroyListeners = this._injector.get(PLATFORM_DESTROY_LISTENERS, null);
+    if (destroyListeners !== null && destroyListeners.size > 0) {
```

Comment:
Nit: 
```suggestion
    if (destroyListeners?.size > 0) {
```

Reply:
I prefer the explicitness of `!== null` in these cases. However, given that this is runtime code, I would suggest even limiting this to
```suggestion
    if (destroyListeners) {
```
or even
```suggestion
    destroyListeners?.forEach(listener => listener());
```

Reply:
Thanks for the feedback. I've updated the code to use a short version (the `if (destroyListeners)` one).